website/docs/topics/feature-flags/best-practices-using-feature-flags-at-scale.md

```diff
@@ -0,0 +1,350 @@
+
+Users who are added or removed from your main SSO directory should be added or removed (based on their role) from your feature flag system.  Managing global permissions is complex. Only do it in one place.
+
+### 6. Implement flag approval workflows early.
+
+![feature-flag-approval-workflow](https://github.com/ferrantim/unleash/assets/3068350/38de8897-be16-43af-80d2-a3f0680248ea)
+
+Depending on the industry and legal framework you’re operating in, you’ll need varying approvals for a feature flag change to go into production. You may be able to just roll something out on a whim, or you might need to get it approved by specific stakeholders before releasing it. Your feature flag tool should help you do this, regardless of how much oversight you need.
+
+**Require approvals where necessary**
+
+When legal requirements say that a change must be peer-reviewed before being put into production (often referred to as the ‘[four-eyes principle](https://www.unido.org/overview-member-states-change-management-faq/what-four-eyes-principle)' or the ‘two-person rule’), it should be easy to do. You should be able to add all the changes you want to make to all the relevant flags and present them as a group, similar to making a pull request on GitHub.
+
+It should be easy for the reviewers to see what changes would be made and what the configuration would look like after the change. 
+
+But this isn’t just useful for legal reasons. The process of submitting something for review is useful in and of itself. 
+
+Review flag changes before they take effect
+
+Even if you don’t require anyone else's approval before making changes in production environments, there are still benefits to grouping your changes together and reviewing them before actually making them:
+
+1. You can make multiple changes at the same time. If you have a set of changes across flags that should change in tandem, you can make sure they update at the same time, avoiding awkward in-between states.
+2. You avoid making changes by mistake. If your cat walks across the keyboard, you can feel safe that they won’t accidentally expose your unfinished code to everyone. 
+3. You get to review the changes before they go out. You get the full overview of the changes you’re making. You can double-check that the changes will have the effect you want.
+
+
+## Instrumenting your code and managing flag lifecycle
+
+Flags live in code, and hundreds of small choices go into instrumenting flags into that code.  Here are some tips we’ve picked up to improve the maintainability and performance of large-scale feature flag systems.
+
+### 7. Define flags in your code at the highest level of abstraction.
+
+![evaluate-flags-high-in-code](https://github.com/ferrantim/unleash/assets/3068350/d355080c-3537-4106-ac7b-3a761ce3ac50)
```

Comment:
Assuming the images has moved to `/img/evaluate-flags-at-the-edge.png`

```suggestion
<Figure img="/img/evaluate-flags-at-the-edge.png" caption="Instead of evaluating a flag in a back-end service and then dynamically changing the behavior in that service (either by calling out to other services or changing how the service works), consider evaluating the flag as far out in the stack as you can and deciding which service to call based on the flag value."/>
```